### ISSUE TITLE
Getting error "Supplied code must return or export a function." on new deployments

### ISSUE THREAD

**Issue by baskin** [auth0/wt-cli#227]
Team,
I recently started getting the error in runtime 
```
{
  "code": 400,
  "message": "Invalid webtask code",
  "error": "Supplied code must return or export a function."
}
```
when I invoke the webtask.
The webtask duly has the structure
```
const webtaskApi = module.webtask;
module.exports = function(e) {
}
```
What could be wrong?

**Comment by NotMyself** [comment]
Hello @baskin,

Your code appears to follow the single parameter programming model described [here](https://webtask.io/docs/model). Although, it will fail because the code never calls the callback function. So execution will timeout.

The error you are getting is not related to the timeout issue though. It appears that the code you have deployed does not follow any of the programming models described in the documentation.

How are you deploying the code? If you are using the CLI, you can issue this command to view the code that has been deployed: `wt inspect <TASK_NAME> --fetch-code`.

**Comment by baskin** [comment]
Thanks @NotMyself for looking into this.

No. The actual code follows the prescribed programming model.
```
function (context, cb) {}
```
What I posted above was the result of minification caused by the `--bundle-minify` option in wt-cli

I use 
`wt create build/webtask/get-story-details --bundle --bundle-minify --secrets-file secrets`
to deploy the function.

----
I think the error code `"Supplied code must return or export a function"` is as a result of some kind of "catch all" handler in webtask.
Is it possible to see what inner exceptions are generated when the task is invoked? A sample way to invoke it is
```
curl -X GET https://wt-21a69ff8241911e6f1bb310eb1a1900d-0.sandbox.auth0-extend.com/get-story-details
{
  "code": 400,
  "message": "Invalid webtask code",
  "error": "Supplied code must return or export a function."
}
```

**Comment by NotMyself** [comment]
I do not seem to be able to reproduce the issue you are describing, here are the exact steps I am doing:

1. `echo "module.exports = function (cb) {cb(null, 'Hello');}" > task.js`
1. `wt create task.js --bundle --bundle-minify`
1.  `wt inspect task --fetch-code`
1. `curl -X GET https://<CONTAINER>.sandbox.auth0-extend.com/task`

![example](https://user-images.githubusercontent.com/73120/45638233-7443e200-ba61-11e8-913b-538f8c906350.gif)

Can you try this? Also, can you give me detailed steps to reproduce the issue?

**Comment by baskin** [comment]
> Can you try this? Also, can you give me detailed steps to reproduce the issue?

This works.

@NotMyself  However the following doesn't (bundling with minification)

1. code
```
> cat build/test.js 
import { Foo } from "./foo";
module.exports =
    function (context, cb) {
        let a = new Foo();
        cb(null, a);
    };

> cat build/foo.js 
export class Foo {
}
```
2. bundle and upload
`> wt create build/test.js --bundle --bundle-minify`

3. and the error
```
> curl -X GET https://wt-21a69ff8241911e6f1bb310eb1a1900d-0.sandbox.auth0-extend.com/test
{
  "code": 400,
  "message": "Invalid webtask code",
  "error": "Supplied code must return or export a function."
}
```

4. If however, I try again, this time w/o the minify option, it works!
```
> wt create build/test.js --bundle 
> curl -X GET https://wt-21a69ff8241911e6f1bb310eb1a1900d-0.sandbox.auth0-extend.com/test
{}
```

**Comment by baskin** [comment]
wt-cli@10.0.3
webtask-bundle@3.2.2
webtask-runtime@1.7.1

```
└─┬ wt-cli@10.0.3
  ├── async@2.6.1
  ├── auth0-hooks-templates@1.2.2
  ├── bluebird@2.11.0
  ├── boom@2.10.1
  ├── bunyan-prettystream@0.1.3
  ├── chalk@1.1.3
  ├── concat-stream@1.6.2
  ├── cron-parser@0.6.4
  ├── debug@2.6.9
  ├── dedent@0.7.0
  ├── dotenv@1.2.0
  ├── filewatcher@3.0.1
  ├── jws@3.1.5
  ├── jwt-decode@1.5.1
  ├── linewrap@0.2.1
  ├── lodash@4.17.11
  ├── moment-timezone@0.5.21
  ├── open@0.0.5
  ├── opn@4.0.2
  ├── pad@1.2.1
  ├── promptly@0.2.1
  ├── sandboxjs@4.2.2
  ├── semver@5.5.1
  ├── structured-cli@1.0.5
  ├── superagent@3.8.3
  ├── superagent-proxy@1.0.3
  ├── through2@2.0.3
  ├── update-notifier@0.6.3
  ├── webtask-bundle@3.2.2
  ├── webtask-runtime@1.7.1
  └── wreck@10.0.0

```

**Comment by NotMyself** [comment]
Just to be complete, what version of node are you running? `node -v`

**Comment by baskin** [comment]
node -v
v7.10.0

**Comment by NotMyself** [comment]
OK, I was able to reproduce the issue using node 7.

![screen shot 2018-09-17 at 1 51 00 pm](https://user-images.githubusercontent.com/73120/45649702-f55ea180-ba80-11e8-8e22-1463035f511f.png)

I would recommend that you upgrade to node 8 LTS.

**Comment by baskin** [comment]
Any particular version of node 8?
I just tried with v8.0.0. Doesn't work. Same error.

```
> n
node/8.0.0
> node -v
v8.0.0
> wt create build/test.js --bundle --bundle-minify                                       
Webtask created
> curl -X GET https://wt-21a69ff8241911e6f1bb310eb1a1900d-0.sandbox.auth0-extend.com/test
{
  "code": 400,
  "message": "Invalid webtask code",
  "error": "Supplied code must return or export a function."
}
```


Also, I've been using the above commands for months. The only thing that has changed is that I upgrade my wt-cli to the latest version (from v9.4). However downgrading back did not fix the issue either.

**Comment by NotMyself** [comment]
I am using v8.11.4

![screen shot 2018-09-17 at 2 15 05 pm](https://user-images.githubusercontent.com/73120/45650754-2ab8be80-ba84-11e8-8b87-7aca0749cf0b.png)

But anything v8 marked LTS should work. If you updated to the same version I am, I would encourage you to wipe out your global node modules are reinstall them.

**Comment by baskin** [comment]
> I would encourage you to wipe out your global node modules are reinstall them.

Ok, thanks will try.

**Comment by NotMyself** [comment]
Hold on @baskin. I am able to reproduce this with node 8.12 using a clean install. Let me dig into this a bit more.

**Comment by NotMyself** [comment]
OK, I have a reproducible case. I am going to log this as a bug. Thanks for working through it with me.

**Comment by baskin** [comment]
Hi @NotMyself ,

Can you help me get a workaround for this, in the meantime?
I've tried around now with different older versions of node, wt-cli, webpack etc.. but wasn't able to make it work.

Any clues of what I could try next would be helpful too. 

Thanks.

**Comment by NotMyself** [comment]
@baskin 

This issue should now be resolved with the release of wt-cli v11.0.0.

**Comment by baskin** [comment]
Thanks a bunch!!
